test/isEmpty.js

```diff
@@ -0,0 +1,138 @@
+'use strict'
+
+/* eslint-env mocha */
+
+const assert = require('assert')
+const { isEmpty } = require('../index')
+
+const _toArgs = (array) => {
+  const wrapper = function () {
+    return arguments
+  }
+
+  return wrapper.apply(undefined, array)
+}
+
+describe('isEmpty', () => {
+  it('should return true for true boolean', () => {
+    assert.strictEqual(isEmpty(true), true)
+  })
+
+  it('should return true for false boolean', () => {
+    assert.strictEqual(isEmpty(false), true)
+  })
+
+  it('should return true for NaN', () => {
+    assert.strictEqual(isEmpty(NaN), true)
+  })
+
+  it('should return true for RegExp', () => {
+    assert.strictEqual(isEmpty(/x/), true)
+  })
+
+  it('should return true for Symbol', () => {
+    assert.strictEqual(isEmpty(Symbol('a')), true)
+  })
+
+  it('should return true for null', () => {
+    assert.strictEqual(isEmpty(null), true)
+  })
+
+  it('should return true for undefined', () => {
+    assert.strictEqual(isEmpty(undefined), true)
+  })
+
+  it('should return true for empty args', () => {
+    assert.strictEqual(isEmpty(), true)
+  })
+
+  it('should return true for 0 Buffer', () => {
+    assert.strictEqual(isEmpty(Buffer.alloc(0)), true)
+  })
+
+  it('should return false for 1 Buffer', () => {
+    assert.strictEqual(isEmpty(Buffer.alloc(1)), false)
+  })
+
+  it('should return true for empty string', () => {
+    assert.strictEqual(isEmpty(''), true)
+  })
+
+  it('should return false for string', () => {
+    assert.strictEqual(isEmpty('test'), false)
+  })
+
+  it('should return true for empty object', () => {
+    assert.strictEqual(isEmpty({}), true)
+  })
+
+  it('should return false for non-empty object', () => {
+    assert.strictEqual(isEmpty({ a: 0 }), false)
+  })
+
+  it('should return false for object that has `length=0` property', () => {
+    assert.strictEqual(isEmpty({ length: 0 }), false)
+  })
+
+  it('should return false for object that has `length="0"` property', () => {
+    assert.strictEqual(isEmpty({ length: '0' }), false)
+  })
+
+  it('should return true for negative lengths as array-like', () => {
```

Comment:
should we add also tests for class instances? e.g. `class Foo {}; const obj = new Foo()`...

Reply:
no need as class is syntactic sugar for prototypes
but I can add it for better readability of the tests
would you like?

Reply:
yes please, let's include them for better long term support